packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/shared/base-load-balancer.ts

```diff
@@ -283,9 +281,13 @@ export abstract class BaseLoadBalancer extends Resource {
     );
 
     // make sure the bucket's policy is created before the ALB (see https://github.com/aws/aws-cdk/issues/1633)
-    // at the L1 level to avoid creating a circular dependency (see https://github.com/aws/aws-cdk/issues/27528)
-    const lb = this.node.defaultChild as CfnLoadBalancer;
-    lb.addDependency(bucket.node.defaultChild as s3.CfnBucket);
+    // at the L1 level to avoid creating a circular dependency (see https://github.com/aws/aws-cdk/issues/27528
+    // and https://github.com/aws/aws-cdk/issues/27928)
+    const lb = this.node.defaultChild;
+    const bucketPolicy = bucket.policy?.node.defaultChild;
+    if (lb && bucketPolicy && CfnResource.isCfnResource(lb) && CfnResource.isCfnResource(bucketPolicy)) {
+      lb.addDependency(bucketPolicy);
+    }
```

Comment:
we started off with this (before #27558)

```ts
    this.node.addDependency(bucket);
```

Now, we are (among other changes) only adding the dependency if the policy exists. So I am concerned about the situation where the bucket does not have a policy. In this situation, it seems we used to add `this.node.addDependency(bucket)`, but now skip adding the dependency altogether. 

@lpizzinidev is that something you've considered?

Reply:
Thanks for the feedback.

If the bucket does not have a policy, then adding a dependency on it would not create a circular dependency.
Are there cases where we want a dependency on the bucket without the policy?
I assume that, if the bucket does not have a policy attached by CDK, the policy must already exist, otherwise logs would not be delivered to S3.

Reply:
Gotcha. We were depending on the bucket before but really only cared about depending on the policy. Thats the nuance I was missing.